### ISSUE TITLE
Zelda: OoT missing fences

### ISSUE THREAD

**Issue by loganmc10** [gonetz/GLideN64#1575]
You can see the original report of this issue (including mupen64plus save state) here:

https://github.com/m64p/mupen64plus-GLideN64/issues/13

I verified the issue, and bisected it to this commit: 187f9ef390052906e3516e1aaa71078125d099de

**Comment by gonetz** [comment]
Hmm. Could you load that save and check in debugger, how ucode is detected? It must have type = F3DEX2.

**Comment by olivieryuyu** [comment]
there is a savestate in the github of m64p/mupen64plus-GLideN64#13

**Comment by loganmc10** [comment]
I see this:
```
RSP Gfx ucode F3DZEX.NoN  fifo 2.06H Yoshitaka Yasumoto 1998 Nintendo.
```

**Comment by loganmc10** [comment]
And sorry yeah, the value of type is detected as 2 / F3DEX2

**Comment by gonetz** [comment]
Hmm, interesting. I thought that Zelda OOT ucode is the same as standard F3DEX2, at least on HLE level. That is it uses the same commands set as F3DEX2. I was wrong.

Zelda OOT uses F3DZEX 2.0H. Zelda MM uses F3DZEX 2.08I

Long ago we found that F3DEX2 command F3DEX2_BRANCH_Z (opcode 0x04) works incorrect for Zelda MM. We found, what command 0x04 for Zelda MM does and implemented it. I created special ucode F3DEX2MM, which has the same set of commands as F3DEX2, except for 0x04, which has special implementation F3DEX2MM_Branch_W. I changed ucode detection in  GBI.cpp, so F3DEX2MM selected for Zelda MM. I made a mistake, and F3DEX2MM set for Zelda OOT too. I noticed that mistake when added support for 0xD5 command and set F3DEX2 for Zelda OOT. As the result Zelda OOT is broken.

I checked how  Zelda OOT calls 0x04 command and found that it is not F3DEX2_BRANCH_Z , but the same F3DEX2MM_Branch_W as in Zelda MM ucode. Surprise.
